### ISSUE TITLE
ext/net: SSL can panic if tokio coop budget exceeded

### ISSUE THREAD

**Issue by NetOpWibby** [denoland/deno#21762]
```
A client connection error occurred; reconnecting because of "waitUntilAvailable=30000". ClientConnectionClosedError: network error: ConnectionReset: Connection reset by peer (os error 54)
    at RawConnection._onError (https://deno.land/x/edgedb@v1.4.1/_src/rawConn.ts:159:20)
    at TLSSocket.emit (https://deno.land/std@0.177.0/node/_events.mjs:379:28)
    at https://deno.land/x/edgedb@v1.4.1/_src/adapter.deno.ts:308:16
    at eventLoopTick (ext:core/01_core.js:181:11) {
  source: ConnectionReset: Connection reset by peer (os error 54)
    at async Object.connectTls (ext:deno_net/02_tls.js:35:51) {
    name: "ConnectionReset",
    code: "ECONNRESET"
  }
}

============================================================
Deno has panicked. This is a bug in Deno. Please report this
at https://github.com/denoland/deno/issues/new.
If you can reliably reproduce this panic, include the
reproduction steps and re-run with the RUST_BACKTRACE=1 env
var set and include the backtrace in your report.

Platform: macos aarch64
Version: 1.39.1
Args: ["deno", "run", "--allow-env", "--allow-net", "--allow-read", "--import-map", "import_map.json", "--unstable", "main.ts"]

thread 'main' panicked at /Users/brew/Library/Caches/Homebrew/cargo_cache/registry/src/index.crates.io-6f17d22bba15001f/rustls-tokio-stream-0.2.16/src/stream.rs:626:15:
internal error: entered unreachable code
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
[E] [daem] app crashed - waiting for file changes before starting ...
```

## Reference

The service I'm building has two parts: API and app.

The API is Deno and GraphQL. It receives a GraphQL query from the app and also queries remote APIs before sending data back to the app.

The app is SvelteKit and GraphQL.

**Comment by NetOpWibby** [comment]
I'm not sure where to put `RUST_BACKTRACE=1`. I thought putting it in my `scripts.json` with `"env": "RUST_BACKTRACE=1",` would work (I'm using denon) but alas.

**Comment by bartlomieju** [comment]
CC @mmastrac

**Comment by mmastrac** [comment]
I'll take a look at this. You should be able to run your code like so:

```
RUST_BACKTRACE=1 denon run ...
```

The error appears to point to a line that should not be throwing that error.

**Comment by NetOpWibby** [comment]
For the sake of completeness, here's a stack trace:

```
============================================================
Deno has panicked. This is a bug in Deno. Please report this
at https://github.com/denoland/deno/issues/new.
If you can reliably reproduce this panic, include the
reproduction steps and re-run with the RUST_BACKTRACE=1 env
var set and include the backtrace in your report.

Platform: macos aarch64
Version: 1.39.1
Args: ["deno", "run", "--allow-env", "--allow-net", "--allow-read", "--import-map", "import_map.json", "--unstable", "main.ts"]

thread 'main' panicked at /Users/brew/Library/Caches/Homebrew/cargo_cache/registry/src/index.crates.io-6f17d22bba15001f/rustls-tokio-stream-0.2.16/src/stream.rs:626:15:
internal error: entered unreachable code
stack backtrace:
   0: _rust_begin_unwind
   1: core::panicking::panic_fmt
   2: core::panicking::panic
   3: <rustls_tokio_stream::stream::TlsStream as tokio::io::async_read::AsyncRead>::poll_read
   4: deno_net::ops_tls::TlsStreamResource::read::{{closure}}
   5: <deno_net::ops_tls::TlsStreamResource as deno_core::resources::Resource>::read_byob::{{closure}}
   6: <futures_util::future::future::Map<Fut,F> as core::future::future::Future>::poll
   7: deno_core::ops_builtin::op_read::v8_fn_ptr
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
```

**Comment by NetOpWibby** [comment]
@mmastrac Do you have any insight on what I can do to minimize this issue (until the fix is in Deno proper)?

**Comment by mmastrac** [comment]
This should be fixed in 1.39.2 -- let me know if you are still seeing it.

**Comment by NetOpWibby** [comment]
@mmastrac Uninstalled brew version, installed script version, confirmed this issue is fixed. Thank you so much!
